libndt.cpp

```diff
@@ -1198,6 +1285,22 @@ int Client::connect(Socket fd, const sockaddr *sa, SockLen len) noexcept {
 }
 
 Ssize Client::recv(Socket fd, void *base, Size count) noexcept {
+#if HAVE_OPENSSL
+  if ((impl->settings.proto & protocol::tls) != 0) {
+    if (count > INT_MAX) {
+      set_last_error(OS_EINVAL);
+      return -1;
+    }
+    if (impl->fd_to_ssl.count(fd) <= 0) {
+      set_last_error(OS_EINVAL);
+      return -1;
+    }
+    auto ssl = impl->fd_to_ssl.at(fd);
+    auto rv = ::SSL_read(ssl, base, count);
```

Comment:
One possible issue is that, because of the way in which the multi stream download is implemented, a renegotiation during the download is not reflected into the flags that we actually check with `select()`.